SignalServiceKit/src/Messages/OWSBatchMessageProcessor.m

```diff
@@ -458,6 +458,12 @@ - (instancetype)initWithPrimaryStorage:(OWSPrimaryStorage *)primaryStorage
 
     _processingQueue = processingQueue;
 
+    [AppReadiness runNowOrWhenAppDidBecomeReady:^{
+        if (CurrentAppContext().isMainApp) {
```

Comment:
Some components (like this one) only need to do their "on startup" work in the main app.

Reply:
Good point. Made the same change in `OWSMessageReceiver`.